Vikta-UI-testing-sample/src/test/java/com/griddynamics/qa/vikta/uitesting/sample/stepsDefinitions/RegistrationSteps.java

```diff
@@ -15,58 +16,54 @@ public class RegistrationSteps extends BaseSteps {
   private static String SUCCESSFUL_REGISTRATION_MESSAGE_PREFIX =
     "User has been registered successfully: ";
 
+  private static String FAILED_REGISTRATION_MESSAGE_PREFIX =
+    "There is already a user registered with the loginname provided";
+
   public RegistrationSteps(WebDriver driver) {
     super(driver);
   }
 
-  public enum FieldName {
-    LOGINNAME,
-    SURNAME,
-    FIRSTNAME,
-    PATRONIM,
-    PASSWORD,
-  }
-
   @Step
   public void openRegistrationPage() {
     getDriver().get(getData().registrationPageUrl());
   }
 
   @Step
-  public String typeRandomValueInto(FieldName fieldName) {
-    String valueToReturn;
-    switch (fieldName) {
-      case LOGINNAME:
-        valueToReturn = generateRandomString();
-        page().typeInLoginname(valueToReturn);
-        break;
-      case SURNAME:
-        valueToReturn = generateRandomString();
-        page().typeInSurname(valueToReturn);
-        break;
-      case FIRSTNAME:
-        valueToReturn = generateRandomString();
-        page().typeInFirstname(valueToReturn);
-        break;
-      case PATRONIM:
-        valueToReturn = generateRandomString();
-        page().typeInPatronim(valueToReturn);
-        break;
-      case PASSWORD:
-        valueToReturn = generateRandomString();
-        page().typeInPassword(valueToReturn);
-        break;
-      //TODO: Add the rest... .
-      default:
-        throw new IllegalArgumentException(
-          "Unsupported Registration page field name: " + fieldName
-        );
-    }
-
-    return valueToReturn;
+  public void fillInRegistrationForm() {
```

Comment:
this method should fill all the required fields. and then you can other methods that fill single fields if you need it. or we can stick with one  step per field filled in. current solution is a bit  misleading, because the method says that  it's filling the form, but  it's actually only couple of fields